SeafFileProvider/SeafEnumerator.m

```diff
@@ -129,11 +138,19 @@ - (NSArray *)getSeafDirProviderItems:(SeafDir *)dir startingAtPage:(NSFileProvid
             [dir reSortItemsByName];
         }
     }
+    
+    NSInteger numPage = [[NSString stringWithUTF8String:[page bytes]] integerValue];
+    NSInteger start = numPage * self.maxItemCount + 1;
+    NSInteger stop = start + (self.maxItemCount - 1);
+    NSInteger counter = 0;
 
     NSMutableArray *items = [NSMutableArray new];
     for (SeafBase *obj in [self getAccessiableSubItems: dir]) {
-        [obj loadCache];
-        [items addObject: [[SeafProviderItem alloc] initWithSeafItem:[SeafItem fromSeafBase:obj]]];
+        counter += 1;
```

Comment:
You can directly iterate thought the array using index, 
NSArray *array .= [self getAccessiableSubItems: dir]
for (idx = start; idx<=stop; ++idx) {
     item = [array objectAtIndex: idx]
}